src/main/groovy/com/calgaryscientific/gradle/VeracodeUploadFile.groovy

```diff
@@ -30,6 +30,26 @@ import groovy.transform.CompileStatic
 
 @CompileStatic
 class VeracodeUploadFile {
+    static void uploadFiles(VeracodeAPI veracodeAPI, File outputFile, Set<File> fileSet, Integer maxTries, Integer waitTime, Boolean delete) {
+        fileSet.each { file ->
+            uploadFile(file, maxTries, waitTime, veracodeAPI, outputFile, false)
+            if (delete) {
+                println("Deleting ${file}")
+                file.delete()
+            }
+        }
+    }
+
+    static void uploadSandboxFiles(VeracodeAPI veracodeAPI, File outputFile, Set<File> fileSet, Integer maxTries, Integer waitTime, Boolean delete) {
+        fileSet.each { file ->
```

Comment:
I'm not comfortable with having too many responsibilities in a single loop, one way to tackle this is:
```
uploadSandboxFiles()
deleteFileIfConfigured()
```

Reply:
I think that in this case creating a method for it is going too far. The uploadFile is already a method call.

```
            uploadFile(file, maxTries, waitTime, veracodeAPI, outputFile, true)
            if (delete) {
                println("Deleting ${file}")
                file.delete()
            }
```

vs

```
            uploadFile(file, maxTries, waitTime, veracodeAPI, outputFile, true)
            deleteFileIfConfigured(file, delete)
```